api/go.mod

```diff
@@ -1,19 +1,18 @@
 module github.com/ceph/ceph-csi/api
 
 go 1.21
```

Comment:
@iPraveenParihar Can you check and update all go.mod to 1.22 ? 
We seemed to have missed updating this go.mod

Reply:
done